framework/db/SchemaBuilder.php

```diff
@@ -0,0 +1,200 @@
+ * @license http://www.yiiframework.com/license/
+ *
+ * @author Vasenin Matvey <vaseninm@gmail.com>
+ * @since 2.0.5
+ */
+
+namespace yii\db;
+
+use Yii;
+use yii\base\Object;
+
+abstract class SchemaBuilder extends Object
+{
+    protected $schema = null;
+    protected $length = null;
+    protected $isNotNull = null;
+    protected $check = null;
+    protected $default = null;
+
+    public static function primaryKey($length = null)
+    {
+        return static::createDefault(Schema::TYPE_PK, $length);
+    }
+
+    public static function bigPrimaryKey($length = null)
+    {
+        return static::createDefault(Schema::TYPE_BIGPK, $length);
+    }
+
+    public static function string($length = null)
+    {
+        return static::createDefault(Schema::TYPE_STRING, $length);
+    }
+
+    public static function text($length = null)
+    {
+        return static::createDefault(Schema::TYPE_TEXT, $length);
+    }
+
+    public static function smallInteger($length = null)
+    {
+        return static::createDefault(Schema::TYPE_SMALLINT, $length);
+    }
+
+    public static function integer($length = null)
+    {
+        return static::createDefault(Schema::TYPE_INTEGER, $length);
+    }
+
+    public static function bigInteger($length = null)
+    {
+        return static::createDefault(Schema::TYPE_BIGINT, $length);
+    }
+
+    public static function float($precision = null, $scale = null)
+    {
+        return static::createNumeric(Schema::TYPE_FLOAT, $precision, $scale);
+    }
+
+    public static function double($precision = null, $scale = null)
+    {
+        return static::createNumeric(Schema::TYPE_DOUBLE, $precision, $scale);
+    }
+
+    public static function decimal($precision = null, $scale = null)
+    {
+        return static::createNumeric(Schema::TYPE_DECIMAL, $precision, $scale);
+    }
+
+    public static function dateTime($length = null)
+    {
+        return static::createDefault(Schema::TYPE_DATETIME, $length);
+    }
+
+    public static function timestamp($length = null)
+    {
+        return static::createDefault(Schema::TYPE_TIMESTAMP, $length);
+    }
+
+    public static function time($length = null)
+    {
+        return static::createDefault(Schema::TYPE_TIME, $length);
+    }
+
+    public static function date($length = null)
+    {
+        return static::createDefault(Schema::TYPE_DATE, $length);
+    }
+
+    public static function binary($length = null)
+    {
+        return static::createDefault(Schema::TYPE_BINARY, $length);
+    }
+
+    public static function boolean($length = null)
+    {
+        return static::createDefault(Schema::TYPE_BOOLEAN, $length);
+    }
+
+    public static function money($precision = null, $scale = null)
+    {
+        return static::createNumeric(Schema::TYPE_MONEY, $precision, $scale);
+    }
+
+    public function notNull()
+    {
+        $this->isNotNull = true;
+
+        return $this;
+    }
+
+    public function setDefault($default = null)
```

Comment:
Yes. I think it worth doing via `__call()`. Performance isn't concern in this case.